### ISSUE TITLE
Question about processed data

### ISSUE THREAD

**Issue by mobashgr** [fenchri/edge-oriented-graph#35]
Hi,
Thanks for the great work. I have a couple of questions regarding the output of the `BC5CDR processed  training data` document `6287825`  as follows:

1. The gold standard relations are denoted by ` 1:CID:2 R2L` , right? The only relation that is CID and cross is  `1:CID:2 R2L     CROSS   129-130 114-118 D007538 Isoniazid       Chemical        129     130     7       D010523 Diseases of peripheral nerves|peripheral nerve disease|Sensori - motor neuropathy|motor neuropathy|Peripheral neuropathy|sensori - motor neuropathy     Disease 0:17:51:87:90:114       4:20:55:89:92:118  0:1:3:4:5:5 `, which means that this relation is inter-sentence relation between `Isoniazid in sentence 7 `and `sensori- motor neuropathy sentence 5` based on the given node's offset. So technically the annotator inferred this based on the context of the 2 sentences, am I right? I believe you name this as positive relation in your code, true?
2.  Regarding the inter-sentence relations that aren't part of the gold standard are denoted by `1:NR:2  L2R     CROSS ` and intra-sentence relations that aren't part of the gold standard are `1:NR:2  R2L     NON-CROSS  `, right? I believe you name this as `negative relation` in your code, true?
3.  Is there a way to express how you labelled` inter-sentence relations`? In other words to define the edge construction whether it is mention-mention or mention-sentence or mention-entity or sentence-sentence? For example, I am not sure if   `1:NR:2 L2R     CROSS   97-98   152-154 D013831 thiamine        Chemical        97      98      5       D003389 cranial neuropathy      Disease 152     154     8`   is `sentence-sentence`? Can you point me to the part where you implemented the different edge construction approaches?
4. Have you considered entity-entity relations? To be more precise, if, for example, `C1 has a relation with C2  and C2 has a relation with C3, so we can infer that C1 has inter-sentence relation with C3`. I understand that in BC5CDR, you focus on chemical-induced-disease relations, however, for example, if I am interested in chemical-chemical relations, would it be possible to add this to the processed data?

Best,
Ghadeer

**Comment by fenchri** [comment]
Hello, thank you for your interest in this work.

R2L and L2R just denotes if the relation is from the 1st or 2nd argument as it appears in the sentence. In this particular dataset there are only chemical-disease relations, so the first one should always be a chemical and the second a disease. 

If you see CROSS in the format, then this means this is a cross-sentence (inter-sentence) relation, otherwise is will have NON-CROSS. If the label is CID this is a positive relation, if it is NR it is negative (no relation).

Regarding the annotation, it was manual from professional annotators if I am not mistaken. The relations should be inferred from the entire context, but you can check the corpus original paper for additional details.

As mentioned before, in order to determine an inter-sentence relation, you need to find all the mentions of an entity and make sure that there is no pair the resides in the same sentence. Then the distance is calculated based on the closest sentences a mention of the entities appears. In the example you provided, there is 1 mention in sentence 5 for "thiamine" and one mention in sentence 8 for "cranial neuropathy", so this is an inter-sentence relation, marked with CROSS.
Edge construction happens [here](https://github.com/fenchri/edge-oriented-graph/blob/master/src/dataset.py#L143).

To clarify, in the paper a mention corresponds to 1 occurrence of an entity, it can have a different name. 1 entity is more global, all possible mentions are associated with it and it is described via its ID. E.g. `motor neuropathy | motor neuropathy | Peripheral neuropathy | sensori - motor neuropathy` all of these mentions are associated with the entity D010523 and can occur in multiple places in the text.

In this work, we did consider chemical-chemical relations inside the model computations, but we classify only chemical-disease pairs. If you want to change that you will need to modify the [config](https://github.com/fenchri/edge-oriented-graph/blob/master/configs/) file and classify all possible entity pairs.
```
include_pairs: ['Chemical-Disease', 'Chemical-Chemical', 'Disease-Disease', 'Disease-Chemical']
classify_pairs: ['Chemical-Disease']
```
If you see the above examples, we include all possible pairs but we classify only those for which we have labels for.
The data pre-processing will need some minor modification as well, in order to include the labels for additional pairs, e.g. Chemical-Chemical.

Hope the above answer your questions.

Fenia
